File: csharp/087_triangle/Triangle.cs
using System;

public static class Triangle
{
    private static bool IsTriangle(double side1, double side2, double side3)
    {
        double[] arr = {side1, side2, side3};
        Array.Sort(arr);
        return arr[0] > 0.0 && arr[0] + arr[1] >= arr[2];
    }
    
    public static bool IsScalene(double side1, double side2, double side3)
        => IsTriangle(side1, side2, side3) && side1 != side2 && side1 != side3 && side2 != side3;

    public static bool IsIsosceles(double side1, double side2, double side3)
        => IsTriangle(side1, side2, side3) && (side1 == side2 || side1 == side3 || side2 == side3);

    public static bool IsEquilateral(double side1, double side2, double side3) 
        => IsTriangle(side1, side2, side3) && side1 == side2 && side2 == side3;
}

File: javascript/ex_048_triangle/triangle.js
export class Triangle {
  constructor(...sides) {
    [this.a, this.b, this.c] = sides.sort((i1,i2) => i1 - i2);
    this.isTriangle = this.a > 0 && (this.a + this.b) >= this.c;
  }

  get isEquilateral() {
    return this.isTriangle && this.a === this.c;
  }

  get isIsosceles() {
    return this.isTriangle && (this.a === this.b || this.b === this.c);
  }

  get isScalene() {
    return this.isTriangle && !this.isIsosceles;
  }
}
